schemas/webpack.d.ts

```diff
@@ -0,0 +1,364 @@
+        descriptionFiles?: string[];
+        enforceExtension?: boolean;
+        enforceModuleExtension?: boolean;
+        extensions?: string[];
+        fileSystem?: any; // TODO missing type
+        mainFields?: [string | string[]];
+        mainFiles?: string[];
+        moduleExtensions?: string[];
+        modules?: string[];
+        plugins?: any[];
+        resolver?: any; // TODO missing type
+        symlinks?: boolean;
+        unsafeCache?: boolean | Object;
+    }
+
+    interface ResolveAlias {
+        alias?: string;
+        name?: string;
+        onlyModule?: boolean;
+    }
+
+    interface Object {
+        [key: string]: any;
+    }
+
+    type EntryItem = string | string[];
+
+    type OutputLibraryTarget =
+        "var"
+        | "assign"
+        | "this"
+        | "window"
+        | "global"
+        | "commonjs"
+        | "commonjs2"
+        | "commonjs-module"
+        | "amd"
+        | "umd"
+        | "umd2"
+        | "jsonp";
+
+    type Stats =
+        "none"
+        | "errors-only"
+        | "minimal"
+        | "normal"
+        | "verbose";
+
+    type Target =
+        "web"
+        | "webworker"
+        | "node"
+        | "async-node"
+        | "node-webkit"
+        | "electron";
+
+    type Externals = string | {key: string} | ((context: any, request: any, callback: (err: any, result: any)=> any) => void) | RegExp;
+
+    type RuleSetCondition = RegExp | string | Function | RuleSetConditionObject;
+    type RuleSetConditionRecursive = RuleSetCondition | RuleSetCondition[];
+    type RuleSetConditions = RuleSetConditionRecursive[];
+    type RuleSetUse = RuleSetUseItem | RuleSetUseItem[];
+    type RuleSetUseItem = string | RuleSetUseItemObject;
+    type RuleSetQuery = Object | string;
+}
+
+interface Webpack {
+    /**
+     * Set the value of `require.amd` and `define.amd`.
+     */
+    amd?: "require.amd" | "define.amd";
+
+    /**
+     * Report the first error as a hard error instead of tolerating it.
+     */
+    bail?: boolean;
+
+    /**
+     * Cache generated modules and chunks to improve performance for multiple incremental builds.
+     *
+     * Boolean: You can pass `false` to disable it.
+     * Object: You can pass an object to enable it and let webpack use the passed object as cache.This way you can share the cache object between multiple compiler calls.
+     */
+    cache?: (boolean | WebpackType.Object);
+
+    /*
+     The base directory (absolute path!) for resolving the `entry` option. If `output.pathinfo` is set, the included pathinfo is shortened to this directory.
+     */
+    context?: string;
+
+    /**
+     * References to other configurations to depend on.
+     */
+    dependencies?: string[];
+
+    /**
+     * TODO add description
+     */
+    devServer?: WebpackType.Object;
+
+    /**
+     * A developer tool to enhance debugging.
+     * Note: Boolean can be used only with value `false`.
+     */
+    devtool?: string | boolean;
```

Comment:
@mhegazy @danielrosenwasser is there a better solution for this thank you can think of?